src/utilcode/winwrappers.cpp

```diff
@@ -0,0 +1,996 @@
+// Licensed to the .NET Foundation under one or more agreements. 
+// The .NET Foundation licenses this file to you under the MIT license. 
+// See the LICENSE file in the project root for more information. 
+
+#include "stdafx.h"
+#include "windows.h"
+#include "winwrappers.h"
+#include "sstring.h"
+#include "ex.h"
+
+typedef InlineSString<2*MAX_PATH> Win32StackSString;  //2*MAX_PATH - is tunable stack allocated buffer size
+
+class LongFile
+{
+    public:
+#ifndef FEATURE_PAL
+        static SString ExtendedPrefix;
+        static SString DevicePathPrefix;
+        static SString UNCPathPrefix;
+        static SString UNCExtendedPathPrefix;
+        static const WCHAR VolumeSeparatorChar;
+        static const WCHAR LongFile::DirectorySeparatorChar;
+        static const WCHAR LongFile::AltDirectorySeparatorChar;
+#endif //FEATURE_PAL
+        static BOOL IsExtended(SString & path);
+        static BOOL IsUNCExtended(SString & path);
+        static BOOL IsDirectorySeparator(WCHAR c);
+        static BOOL IsRelative(SString & path);
+        static BOOL IsDevice(SString & path);
+
+        static HRESULT NormalizePath(SString& path);
+};
+
+HMODULE
+LoadLibraryExWrapper(
+        LPCWSTR lpLibFileName,
+        HANDLE hFile,
+        DWORD dwFlags
+        )
+{
+    HRESULT hr   = S_OK;
+    HMODULE ret = NULL;
+    DWORD lastError;
+
+    EX_TRY
+    {
+
+        Win32StackSString path(Win32StackSString::Literal, lpLibFileName);
+
+        if (LongFile::IsRelative(path) || SUCCEEDED(LongFile::NormalizePath(path)))
```

Comment:
According to the documentation the path has to be fully qualified for it to search the path. So I think in this case IsRelative is correct.

https://msdn.microsoft.com/en-us/library/windows/desktop/ms684179.aspx